src/libsyntax/tokenstream.rs

```diff
@@ -137,13 +137,8 @@ impl TokenTree {
 /// The goal is for procedural macros to work with `TokenStream`s and `TokenTree`s
 /// instead of a representation of the abstract syntax tree.
 /// Today's `TokenTree`s can still contain AST via `token::Interpolated` for back-compat.
-///
-/// The use of `Option` is an optimization that avoids the need for an
-/// allocation when the stream is empty. However, it is not guaranteed that an
-/// empty stream is represented with `None`; it may be represented as a `Some`
-/// around an empty `Vec`.
 #[derive(Clone, Debug)]
```

Comment:
```suggestion
#[derive(Default, Clone, Debug)]
```
Instead of `fn empty`.

Reply:
Good idea.